src/pdf/popplerapi/PopplerGlibPage.cpp

```diff
@@ -78,3 +80,23 @@ auto PopplerGlibPage::findText(string& text) -> vector<XojPdfRectangle> {
 
     return findings;
 }
+
+auto PopplerGlibPage::getLinks() -> std::vector<Link> {
+    std::vector<Link> results;
+    const double height = getHeight();
+
+    GList* links = poppler_page_get_link_mapping(this->page);
+    for (GList* l = links; l != NULL; l = g_list_next(l)) {
+        auto* link = static_cast<PopplerLinkMapping*>(l->data);
+        if (link->action->type == POPPLER_ACTION_URI && link->action->uri.uri) {
+            GUri* uri = g_uri_parse(link->action->uri.uri, G_URI_FLAGS_NONE, nullptr);
```

Comment:
Would it be safe to store `link->action->uri.uri` as a string? We would be avoiding the need for a URI-parsing library. I think I'm missing something here...

Reply:
If we could, we would need to ensure the following:
* The URI is a valid URI. Does poppler always ensure that the PDF links are valid URIs?
* The URI is properly encoded.